### DISCUSSION TITLE
Doesn't work with XT-4 firmware 2.00

### DISCUSSION THREAD

**Discussion by 93RichardB** [gkoh/furble#45] [Q&A] [answered]
Hi, tried with M5Stick CPlus1.1, X-T4 and firmware 2.00, but doesn't connect.  Any suggestions?

**Comment by gkoh** [comment]
@93RichardB Thanks for the query.
I haven’t had a chance to upgrade to firmware 2.00 yet.
Sounds like I will need to repeat the reverse engineering to get it working again.
Give me some time to find time to investigate.

**Comment by 93RichardB** [comment]
OK, I am very interested in the process too.  I can see the various options to program the M5, but it isn’t clear how you do the reverse engineering to get the Fuji structures.

Richard
Sent from my iPad
________________________________

**Comment by gkoh** [comment]
See here for the protocol part (you need an Android device):
https://github.com/gkoh/furble#protocol-reverse-engineering

After that, we infer structures from the bytes we see over the wire.
There is some educated guessing and experimentation to get something working.

**Comment by 93RichardB** [comment]
Thank you.  Will read.  I have mostly iOS, but I think I have an android tablet somewhere.

Thanks for the replies, and for the project!

Regards

Richard

Richard
Sent from my iPad
________________________________

**Comment by gkoh** [comment, marked as answer]
This is now #46.
